extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/handler/AstroThingHandler.java

```diff
@@ -164,112 +184,123 @@ private void restartJobs() {
-                            JobDataMap jobDataMap = new JobDataMap();
-                            jobDataMap.put(AbstractBaseJob.KEY_THING_UID, thingUid);
-                            jobDataMap.put(AbstractBaseJob.KEY_JOB_NAME, "job-daily");
-
-                            // dailyJob
-                            Trigger trigger = newTrigger().withIdentity("trigger-daily-" + typeId, thingUid)
-                                    .withSchedule(CronScheduleBuilder.cronSchedule("0 0 0 * * ?")).build();
-                            JobDetail jobDetail = newJob(getDailyJobClass())
-                                    .withIdentity("job-daily-" + typeId, thingUid).usingJobData(jobDataMap).build();
-                            quartzScheduler.scheduleJob(jobDetail, trigger);
-                            logger.info("Scheduled astro job-daily-{} at midnight for thing {}", typeId, thingUid);
-
-                            // startupJob
-                            trigger = newTrigger().withIdentity("trigger-daily-startup-" + typeId, thingUid).startNow()
-                                    .build();
-                            jobDetail = newJob(getDailyJobClass()).withIdentity("job-daily-startup-" + typeId, thingUid)
-                                    .usingJobData(jobDataMap).build();
-                            quartzScheduler.scheduleJob(jobDetail, trigger);
-
-                            if (isPositionalChannelLinked()) {
-                                // positional intervalJob
-                                jobDataMap = new JobDataMap();
-                                jobDataMap.put(AbstractBaseJob.KEY_THING_UID, thingUid);
-                                jobDataMap.put(AbstractBaseJob.KEY_JOB_NAME, "job-positional");
-
-                                Date start = new Date(System.currentTimeMillis() + (thingConfig.getInterval()) * 1000);
-                                trigger = newTrigger().withIdentity("trigger-positional", thingUid).startAt(start)
-                                        .withSchedule(simpleSchedule().repeatForever()
-                                                .withIntervalInSeconds(thingConfig.getInterval()))
-                                        .build();
-                                jobDetail = newJob(PositionalJob.class).withIdentity("job-positional", thingUid)
-                                        .usingJobData(jobDataMap).build();
-                                quartzScheduler.scheduleJob(jobDetail, trigger);
-                                logger.info("Scheduled astro job-positional with interval of {} seconds for thing {}",
-                                        thingConfig.getInterval(), thingUid);
-                            }
-                        }
+                        // Execute Positional Job immediately
+                        currentTime = LocalDateTime.now();
+                        cronExpression = new CronExpression(createCronFromTemporal(currentTime));
```

Comment:
See above, this doesn't work. To do this more reliably please call `positionalJob.run()` here directly to get the initial data.